Python/thread_pthread.h

```diff
@@ -356,6 +356,16 @@ PyThread_exit_thread(void)
     pthread_exit(0);
 }
 
+void _Py_NO_RETURN
+_PyThread_hang_thread(void)
+{
+    while (1) { /* loop just in case sigsuspend returns for some reason */
+        sigset_t set; /* we don't want to receive any signal */
+        sigfillset(&set);
+        sigsuspend(&set);
```

Comment:
This code changes the behavior if a program is calling signal.pause() or a variant in a different thread (ex: main thread): this thread can now "eat" the signal silently, no?

Reply:
According to the `signal` man page, blocking a signal does not cause it to be ignored (eaten), it just means it will be sent to a different thread, or if there is no other thread where it is unblocked, it will remain pending until there is a thread that can receive it.

Reply:
I still fear that sigsuspend() can have side effects :-(

Instead, I suggest using a lock and acquire it twice. We can use our portable API for that.

As far as I know, attempting to acquire a lock twice has no known side effect on other threads. Locks (mutex) have issues with fork, but this hang thread must not be inherited by fork which only inherits the thread calling fork() (a child process has a single thread).

Reply:
It isn't actually important one way or another if this hung thread processes signals, so I wouldn't bother disabling them.

A double lock acquire would work, but for consistency this could also just be turned into a while (1) { sleep(equivalent-to-windows-sleepex-INFINITE); } loop to match the windows code.

signal handlers never call into python or touch locks anyways.

Reply:
I agree with this analysis.  But I don't want to get stuck in a code review loop here.  Originally I had an infinite loop calling `pause`, which does not block signals (but gets interrupted by a signal), but there was a request to block signals, so I changed it to this form.  Happy to have it either way.